target/linux/ath79/dts/qca9563_glinet_gl-ar750s.dtsi

```diff
@@ -115,42 +119,51 @@
 				read-only;
 			};
 
-			partition@60000 {
-				compatible = "denx,uimage";
-				label = "firmware";
+			nor_firmware: partition@60000 {
+				label = "nor_firmware";
 				reg = <0x060000 0xfa0000>;
 			};
+
+			nor_kernel: partition_alt@60000 {
+				label = "nor_kernel";
+				reg = <0x060000 0x200000>;
+			};
+
+			nor_reserved: parition_alt@260000 {
+				label = "nor_reserved";
+				reg = <0x260000 0xda0000>;
+			};
```

Comment:
Is this valid? So with nor-nand board, you have both nor_firmware and nor_kernel at the same position?

Reply:
Yes, it is valid to have overlapping or fully-contained MTD partitions. As an "everyday" example, "firmware" gets auto-split on many OpenWrt boards in to "kernel" and "rootfs", and then additionally "rootfs_data".

NOR:

```
[    0.408802] m25p80 spi0.0: w25q128 (16384 Kbytes)
[    0.413754] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.420330] Creating 4 MTD partitions on "spi0.0":
[    0.425295] 0x000000000000-0x000000040000 : "u-boot"
[    0.431197] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.437365] 0x000000050000-0x000000060000 : "art"
[    0.442972] 0x000000060000-0x000001000000 : "firmware"
[    0.451649] 2 uimage-fw partitions found on MTD device firmware
[    0.457765] Creating 2 MTD partitions on "firmware":
[    0.462952] 0x000000000000-0x0000001e0000 : "kernel"
[    0.468756] 0x0000001e0000-0x000000fa0000 : "rootfs"
[    0.474575] mtd: device 5 (rootfs) set to be root filesystem
[    0.481835] 1 squashfs-split partitions found on MTD device rootfs
[    0.488232] 0x000000a80000-0x000000fa0000 : "rootfs_data"
[    0.495636] spi-nand spi0.1: GigaDevice SPI NAND was found.
[    0.501463] spi-nand spi0.1: 128 MiB, block size: 128 KiB, page size: 2048, OOB size: 128
[    0.510268] 1 fixed-partitions partitions found on MTD device spi0.1
[    0.516829] Creating 1 MTD partitions on "spi0.1":
[    0.521827] 0x000000000000-0x000008000000 : "nand_ubi"
```

NOR-NAND:

```
[    0.413786] 6 fixed-partitions partitions found on MTD device spi0.0
[    0.420365] Creating 6 MTD partitions on "spi0.0":
[    0.425333] 0x000000000000-0x000000040000 : "u-boot"
[    0.431240] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.437394] 0x000000050000-0x000000060000 : "art"
[    0.443004] 0x000000060000-0x000001000000 : "nor_firmware"
[    0.449323] 0x000000060000-0x000000260000 : "kernel"
[    0.455164] 0x000000260000-0x000001000000 : "nor_reserved"
[    0.465653] spi-nand spi0.1: GigaDevice SPI NAND was found.
[    0.471467] spi-nand spi0.1: 128 MiB, block size: 128 KiB, page size: 2048, OOB size: 128
[    0.480084] 1 fixed-partitions partitions found on MTD device spi0.1
[    0.486643] Creating 1 MTD partitions on "spi0.1":
[    0.491629] 0x000000000000-0x000008000000 : "ubi"
```